eessi/reframe/eessi_utils/hooks.py

```diff
@@ -77,6 +87,16 @@ def assign_one_task_per_gpu(test: rfm.RegressionTest) -> rfm.RegressionTest:
     or nb of GPUs per node available in this partition (whatever is smallest).
     """
     max_gpus_per_node = utils.get_num_gpus_per_node(test)
+
+    gpus_per_node = test.gpus_per_node
+    if gpus_per_node:
+        if gpus_per_node > max_gpus_per_node:
+            test.skip(
+                f'GPUs per node ({gpus_per_node}) is higher than available ({max_gpus_per_node}) in this partition')
+        max_gpus_per_node = gpus_per_node
+    else:
+        max_gpus_per_node = math.ceil(max_gpus_per_node / test.node_part)
```

Comment:
`test.node_part` is not a standard test attribute. Thus, there is no guarantee that it is set. I'm guessing it is set in the GROMACS test because we run `set_test_scale` before `assign_one_task_per_gpu`. However, if someone writes a test that doesn't invoke `set_test_scale`, this line will fail, won't it?

Reply:
done in https://github.com/EESSI/test-suite/pull/28/commits/a3669cf7ab31958e4644334ef7f8a65f6ef0a41a